src/series/algorithm/waterfall.js

```diff
@@ -0,0 +1,100 @@
+import d3 from 'd3';
+import {noop} from '../../util/fn';
+
+export default function() {
+
+    var xValue = function(d) { return d.date; },
+        yValue = function(d) { return d.close; },
+        finalTotal = function(d) { return d3.time.day.offset(d.date, 1); },
+        totals = noop;
+
+    var waterfall = function(data) {
+        var length = data.length,
+            i,
+            previousEnd = yValue(data[0]),
+            start,
+            end,
+            total,
+            result = [];
+
+        // First value is a total
+        result.push({
```

Comment:
What I mean by 'total' is that the column has its full height, i.e. `y0` is 0. In a sense it also is a total, by virtue of being the first point.